### ISSUE TITLE
20.09

### ISSUE THREAD

**Issue by jamessewell** [erikarvstedt/extra-container#14]
I know it's still beta, but 20.09 is imminent and appears to dislike extra-container (installed from GIT master).

```
[nix-shell:~/containers]$ nixos-version
20.09beta1057.0b8799ecaaf (Nightingale)

[nix-shell:~/containers]$ sudo extra-container create --start --build-args --show-trace <<'EOF'
{
  containers.demo = {
    privateNetwork = true;
    hostAddress = "10.250.0.1";
    localAddress = "10.250.0.2";

    config = { pkgs, ... }: {
      # Reduce evaluation time. Not needed for NixOS ≥ 20.03
      documentation.nixos.enable = false;

      networking.firewall.allowedTCPPorts = [ 50 ];

      systemd.services.hello = {
        wantedBy = [ "multi-user.target" ];
        script = ''
          while true; do
            echo hello | ${pkgs.netcat}/bin/nc -lN 50
          done
        '';
      };
    };
  };
}
EOF

Building containers...
error: while evaluating the attribute 'config.system.build.etc' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/lib/eval-config.nix:64:5:
while evaluating 'evalModules' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:21:17, called from /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/lib/eval-config.nix:58:12:
while evaluating the attribute 'unmatchedDefns' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:234:10, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:59, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:234:16:
while evaluating the attribute 'environment' at undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:369:22, called from undefined position:
while evaluating the attribute 'unmatchedDefns' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:234:10, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:59, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:234:16:
while evaluating the attribute 'systemPackages' at undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:369:22, called from undefined position:
while evaluating the attribute 'unmatchedDefns' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:7:
while evaluating 'mapAttrsToList' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:233:23, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:376:37:
while evaluating 'byName' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:321:25, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:337:22:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:322:22, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:322:9:
value is a list while a set was expected, at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:323:25
```

**Comment by jamessewell** [comment]
20.09 has been released now, but this is still an issue

**Comment by erikarvstedt** [comment]
Fixed in release `0.5`.
